lib/spack/docs/packaging_guide.rst

```diff
@@ -3453,6 +3442,12 @@ directory, including cached virtual indices.
 
 To remove all of the above, the command can be called with ``--all``.
 
+Also, cleans up temporary files for a particular package, by deleting the
+expanded/checked out source code *and* any downloaded archive.  If
+``fetch``, ``stage``, or ``install`` are run again after this, Spack's
+build process will start from scratch.
```

Comment:
Can we keep this extension for another PR? The code here behaves exactly like the code in `develop`, it just merges the two commands. Example:
```console
$ spack clean --stage zlib
==> Cleaning build stage [zlib@1.2.11%gcc@4.8+pic+shared arch=linux-ubuntu14-x86_64 /pgxsxv7]
==> Removing all temporary build stages

$ spack clean --all
==> Removing all temporary build stages
==> Removing cached downloads
==> Removing cached information on repositories
```